### PR TITLE
Automatic Encrypted DMG Support

### PR DESCRIPTION
**Pull request by andrewboyd-ge**
Add the capability to set a decryption password for updating with an encrypted dmg file without prompting the user for a password.

This enables the the ability to update using an encrypted disk image without having to prompt the user for a password. This is useful when access to the application binaries is only available via distribution or from a location in which authentication is required. This can help prevent an attacker from having unwanted access to application binaries for reverse engineering.

This assists with a defense in depth approach to security.

### CODE REVIEW

**Comment by kainjow** [comment]
IMO should resubmit this without your merge commit.

**Comment by andrewboyd-ge** [comment]
Will reopen another with the merge commit

**Comment by pornel** [comment]
Thank you for the pull request. Can you explain what is the threat model? Who is the attacker, who's trusted?

File: Sparkle/SUDiskImageUnarchiver.m [hunk 1]
```diff
@@ -70,8 +70,21 @@ - (void)extractDMGWithPassword:(NSString *)__unused password
...
-
+        NSMutableArray *arguments = [@[@"attach", self.archivePath, @"-mountpoint", mountPoint, /*@"-noverify",*/ @"-nobrowse", @"-noautoopen"] mutableCopy];
+    
+        if (self.decryptionPassword) {
+            NSMutableData *passwordData = [[self.decryptionPassword dataUsingEncoding:NSUTF8StringEncoding] mutableCopy];
+            // From the hdiutil docs:
+            // read a null-terminated passphrase from standard input
+            //
+            // Add the null terminator, then the newline
+            [passwordData appendData:[NSData dataWithBytes:"\0\n" length:1]];
```
Comment (pornel) [inline Sparkle/SUDiskImageUnarchiver.m]: that's 2 characters and length:1

**Comment by pornel** [comment]
#685

**Pull request closed without merging**